### PR TITLE
feat: better error handling for packer

### PR DESCRIPTION
**Pull request by kylo252**
# Description

- better error handling for packer
- use packer in protected mode

### CODE REVIEW

File: .github/workflows/install.yaml [hunk 1]
```diff
@@ -33,12 +33,6 @@ jobs:
         run: |
           ./utils/installer/install.sh --local --no-install-dependencies
 
-      - name: Test LunarVim PackerCompile
-        run: if "$HOME"/.local/bin/lvim --headless +PackerCompile -c ':qall' 2>&1|grep -q 'Error'; then false; fi
-
-      - name: Test LunarVim Health
-        run: if "$HOME"/.local/bin/lvim --headless +checkhealth -c ':qall' 2>&1|grep -q 'Error'; then false; fi
```
Comment (abzcoding) [inline .github/workflows/install.yaml:40]: is there a reason we are removing health check and packercompile test from our CI?

**Comment by abzcoding** [comment]
the only issue that i get is this
Screen Shot 2021-10-31 at 7 07 55 PM
Screen Shot 2021-10-31 at 7 08 03 PM

**Comment by kylo252** [comment]
> the only issue that i get is this Screen Shot 2021-10-31 at 7 07 55 PM Screen Shot 2021-10-31 at 7 08 03 PM

This means it actually intercepted an error, in theory at least. Do you see any warnings from Packer?

**Comment by abzcoding** [comment]
> This means it actually intercepted an error, in theory at least. Do you see any warnings from Packer?

i've not seen any errors 🤔 
maybe i need to `LvimCacheReset` or sth 🤷

**Comment by kylo252** [comment]
> i've not seen any errors 🤔 maybe i need to `LvimCacheReset` or sth 🤷

when are you getting that error exactly? btw, I finally fixed/added `LvimReload` 😄

**Comment by abzcoding** [comment]
> > i've not seen any errors 🤔 maybe i need to `LvimCacheReset` or sth 🤷
> 
> when are you getting that error exactly? btw, I finally fixed/added `LvimReload` 😄

basically the Packer floating window comes up and does its thing, and i get that harmless error in the side 😅

**Comment by kylo252** [comment]
> basically the Packer floating window comes up and does its thing, and i get that harmless error in the side 😅

make sure you have set

```lua
lvim.log.level = "trace"
```

some tests to try:
- `PackerSync`, does it produce a different output than `LvimReload`?
- `PackerCompile`, does it cause any errors?
- `:lua require("lvim.plugin-loader"):sync()` <-- this will probably trigger an error

**Comment by kylo252** [comment]
I think it should be solved now, I was doing some dumb things that we shall not make any mention of..

**Comment by abzcoding** [comment]
> I think it should be solved now, I was doing some dumb things that we shall not make any mention of..

yeah its fixed now

**Comment by kylo252** [comment]
> yeah its fixed now

I decided to re-adjust the scope of the PR, only expose/use the protected mode for core-plugins. That's why I reverted the which-key bindings. 

Now I need to figure out how to fix the race-condition between `PackerSync` and `PackerCompile` for the installer.

**Pull request merged**